git_machete/client.py

```diff
@@ -710,6 +710,9 @@ def traverse(
 
             needs_slide_out: bool = self.__is_merged_to_upstream(
                 branch, opt_no_detect_squash_merges=opt_no_detect_squash_merges)
+            if branch in self.annotations:
+                if self.annotations[branch].qualifiers.slide_out is not None:
+                    needs_slide_out = self.annotations[branch].qualifiers.slide_out
```

Comment:
I'd suggest just sth like:
```suggestion
                if needs_slide_out and branch in self.annotations:
                    needs_slide_out = self.annotations[branch].qualifiers.slide_out
```

Reply:
A different logic can ofc be applied... but I'd then see it applied consistently for all three qualifiers 🤔

Reply:
@PawelLipski Hmm, this implementation would also fail in the example case in the comment above 🆙

Reply:
> @PawelLipski Hmm, this implementation would also fail in the example case in the comment two comments above 🆙

Pls re-specify the exact case and what would be the actual vs expected behavior 🤔

Reply:
Ok, lets consider 3 cases for qualifiers values (no qualifiers, qualifiers but without `slide-out=no`, `slide-out=no` qualifier) and 2 cases for initial value of `needs_slide_out`:
`needs_slide_out`=`True`:
    - no qualifiers
       - current implementation:
          -  no change in the value of `needs_slide_out` -> correct
       - your proposed implementation:
          -  no change in the value of `needs_slide_out` -> correct
    - `rebase=no` qualifier
       - current implementation:
          -  no change in the value of `needs_slide_out` -> correct
       - your proposed implementation:
          - would change `needs_slide_out` from `True` to `True` -> correct
    - `slide-out=no` qualifier
       - current implementation:
          -  would change the value of `needs_slide_out` from `True` to `False` -> correct
       - your proposed implementation:
          - would change `needs_slide_out` from `True` to `False` -> correct
`needs_slide_out`=`False`:
   easier example - no need to dig deeper, both implementations are correct.

So to sum up, both are correct (my bad, i though your proposition fails in one case but turns out it doesn't), but taking into account that your proposition follows the same pattern as already existing implementation for rebase/push qualifiers - it will be used instead of my current one.

Reply:
Ach, formatting doesn't work, i'll add the screenshot below:
![image](https://user-images.githubusercontent.com/86779115/215520969-447d4151-021b-42df-a487-f9299fdc4f69.png)

Reply:
ok! gr8